Ops/Expr/GenXML.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace PSI;
+
+using System.Xml.Linq;
+
+// An basic XML code generator, implemented using the Visitor pattern
+public class ExprXMLGen : Visitor<XElement> {
+   XElement? mElem;
+
+   public override XElement Visit (NLiteral lit)
+      => New ("Literal").Set ("Value", lit.Value.Text).Set ("Type", lit.Type);
+
+   public override XElement Visit (NIdentifier ident)
+      => New ("Ident").Set ("Name", ident.Name).Set ("Type", ident.Type);
+
+   public override XElement Visit (NUnary unary) {
+      var a = unary.Expr.Accept (this);
+      mElem = New ("Unary").Set ("Op", unary.Op.Kind).Set ("Type", unary.Type);
+      mElem.Add (a);
+      return mElem;
+   }
+
+   public override XElement Visit (NBinary binary) {
+      var left = binary.Left.Accept (this); var right = binary.Right.Accept (this);
+      mElem = New ("Binary").Set ("Op", binary.Op.Kind).Set ("Type", binary.Type);
+      mElem.Add (left); mElem.Add (right);

```

Comment:
Just like you have New and Set, why not have an Add routine too, so you could write in a single line:
`return New ("Unary").Set ("Op", unary.Op.Kind).Set ("Type", unary.Type).Add (unary.Expr.Accept (this));`